Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Templates/LambdaFunctionTemplate.tt

```diff
@@ -79,8 +79,16 @@ namespace <#= _model.LambdaMethod.ContainingNamespace #>
     if (_model.LambdaMethod.Events.Contains(EventType.API))
     {
         var parameters = string.Join(", ", _model.LambdaMethod.Parameters.Select(p => p.Name));
-        var restApiAttribute = _model.LambdaMethod.Attributes.FirstOrDefault(att => att.Type.FullName == TypeFullNames.RestApiAttribute);
+        var restApiAttribute = _model.LambdaMethod.Attributes.FirstOrDefault(att => att.Type.FullName == TypeFullNames.RestApiAttribute) as AttributeModel<RestApiAttribute>;
         var httpApiAttribute = _model.LambdaMethod.Attributes.FirstOrDefault(att => att.Type.FullName == TypeFullNames.HttpApiAttribute) as AttributeModel<HttpApiAttribute>;
+
+        if (restApiAttribute != null && httpApiAttribute != null)
```

Comment:
Not that having both makes a lot of sense but I suspect that we have encoded into our logic that you can only have one event source per LambdaFunction. We should revisit this as we add more event sources to allow users to potentially have more then one event source. Imagine the scenario we have a method that for an API it gets is payload from an the response body but in an SQS event source it gets its payload from the SQS queue. We can tackle this later as we flesh out other event sources but just keep in mind this is a common request we get to have a single Lambda function with multiple event sources.

Reply:
The current design has been kept flexible and it can be changed to allow multiple events on a Lambda function. Once we add a new event, we can have a better design to move forward.

I'm even inclined over removing `EventType` enum and only rely on attributes (they were not modeled previously) which gives better flexibility to incorporate multiple events.